### PR TITLE
Tektro sequence

### PR DESCRIPTION
**Pull request by leghtas**

### CODE REVIEW

File: ecpy_hqc_legacy/instruments/drivers/dll/alazar935x.py [hunk 1]
```diff
@@ -98,9 +98,8 @@ def configure_board(self):
 
         board.setExternalTrigger(ats.DC_COUPLING, ats.ETR_5V)
 
-        triggerDelay_sec = 0.
-        triggerDelay_samples = int(triggerDelay_sec * self.samples_per_sec +
-                                   0.5)
+        triggerDelay_sec = 0
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/dll/alazar935x.py:101]: Please avoid camelCase like name !

File: ecpy_hqc_legacy/instruments/drivers/driver_tools.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def __set__(self, obj, value):
             super(instrument_property, self).__set__(obj, value)
 
 
-def secure_communication(max_iter=2):
+def secure_communication(max_iter=10):
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/driver_tools.py]: why this change ?

File: ecpy_hqc_legacy/instruments/drivers/driver_tools.py [hunk 3]
```diff
@@ -133,7 +133,6 @@ def wrapper(self, *args, **kwargs):
                     if i == max_iter:
                         raise
                     else:
-                        print(format_exc())
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/driver_tools.py:136]: We should probably log the exception.

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py [hunk 4]
```diff
@@ -56,6 +56,25 @@ def clear_sequence(self):
...
+    @secure_communication()
+    def set_sequence_pos(self, name, position):
+        """Sets the sequence index position to waveform name.
+
+        """
+        self._AWG.clear_output_buffer()
+        try:
+            current_length = int(self._AWG.ask("SEQuence:LENGth?"))
+        except:
+            print('Could not read current_length, assuming 0')
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: I would prefer logging over printing here.

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py [hunk 5]
```diff
@@ -556,6 +646,7 @@ def sampling_frequency(self, value):
     def running(self):
         """Run state getter method
         """
+        self.clear_output_buffer()
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: Why so many calls to this method ? I guess it should be called in reopen_connection or in some error handling function.

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py [hunk 6]
```diff
@@ -604,25 +697,34 @@ def run_mode(self, value):
...
+            if self.run_mode[:-1] != 'SEQ':
                 raise InstrIOError(cleandoc('''Instrument did not set
-                                            correctly the run mode'''))
+                                            correctly the run mode.'''))
         else:
             mess = fill(cleandoc('''The invalid value {} was sent to
                                  run mode method''').format(value), 80)
             raise VisaTypeError(mess)
+
+    def delete_all_waveforms(self):
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py:776]: Missing docstring

File: ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py [hunk 7]
```diff
@@ -377,7 +377,7 @@ def perform(self):
 
         waiting_time = self.channel_driver.sweep_time
         self.driver.fire_trigger(self.channel)
-        time.sleep(waiting_time)
+        time.sleep(1.5*waiting_time)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py]: Why this change ?

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+from ecpy.tasks.api import (InstrumentTask, InterfaceableTaskMixin,
+                            validators)
+
+
+class RunAWGTask(InstrumentTask):
+    """ Task to set AWG run mode
+
+    """
+    switch = Unicode('Off').tag(pref=True, feval=validators.SkipLoop())
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: Missing attributes documentation.

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        if switch is None:
+            switch = self.format_and_eval_string(self.switch)
+
+        if switch == 'On' or switch == 1:
+            self.driver.running = 1
+            self.write_in_database('output', 1)
+        else:
+            self.driver.running = 0
+            self.write_in_database('output', 0)
+        print('RUNNING STATE OK')
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: Either log or remove this.

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py [hunk 10]
```diff
@@ -37,6 +37,8 @@ class DemodSPTask(InstrumentTask):
     average = Enum('No avg', 'Avg before demod',
                    'Avg after demod').tag(pref=True)
 
+    num_loop = Unicode('1').tag(pref=True, feval=VAL_INT)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: Missing attribute docstring

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py [hunk 11]
```diff
@@ -214,21 +224,23 @@ def perform(self):
...
                 ch1_c_t = ch1_i_t + 1j*ch1_q_t
-                chc_c_t = np.transpose(np.transpose(ch1_c_t)/ch2_c)
+                chc_c_t = np.swapaxes(np.swapaxes(ch1_c_t, 0, 2)/ch2_c.T, 0, 2)
                 chc_i_t = np.real(chc_c_t)
                 chc_q_t = np.imag(chc_c_t)
 
                 if not average:
-                    chc_i_t_av = chc_i_t
-                    chc_q_t_av = chc_q_t
+                    chc_i_t_av = np.swapaxes(chc_i_t,0,1)[0]
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: Missing space after the comma

File: ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml [hunk 12]
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015-2016 by EcpyHqcLegacy Authors, see AUTHORS for more details.
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml]: Update copyright to 2017

File: ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+"""
+from __future__ import (division, unicode_literals, print_function,
+                        absolute_import)
+
+from enaml.core.api import Conditional
+from enaml.widgets.api import (GroupBox, Label, Field, ObjectCombo, CheckBox)
+from enaml.stdlib.fields import FloatField
+from enaml.layout.api import factory
+
+from textwrap import fill
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml]: stdlib imports should be before external libraries imports

File: ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml [hunk 14]
```diff
@@ -69,6 +69,13 @@ enamldef DemodSpView(InstrView): view:
                     "then average.\nThis allows to use a ref Iref, Qref "
                     "and correct each I,Q pair by the ref, then average.")
 
+    Label: num_loop:
+        text = 'Number of pulse sequences in loop'
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml:73]: The impact of this parameter needs to be detailed probably in the tooltip of the field

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py [hunk 15]
```diff
@@ -315,13 +318,16 @@ def perform(self):
...
+            self.array_dims = list()
             for i, (l, v) in enumerate(self.saved_values.items()):
                 label = self.format_string(l)
                 value = self.format_and_eval_string(v)
                 if isinstance(value, numpy.ndarray):
                     names = value.dtype.names
-                    self.array_values.add(i)
+                    self.array_values.append(i)
+                    self.array_dims.append(value.ndim)
+                    print(value.shape)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: Please remove

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py [hunk 16]
```diff
@@ -333,50 +339,87 @@ def perform(self):
 
             self.initialized = True
 
-        lengths = set()
+        print(self.array_dims)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: Same here

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py [hunk 17]
```diff
@@ -333,50 +339,87 @@ def perform(self):
...
+                        else:
+                            columns.append(val)
                     else:
-                        columns.append(val)
-                else:
-                    columns.append(numpy.ones(length)*val)
+                        columns.append(numpy.ones(length)*val)
+            else:
+                for i, val in enumerate(values):
+                    print('arghhhhh')
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: WTH ?

**Review by MatthieuDartiailh — CHANGES_REQUESTED** [review]
Not bad but some changes are required.

File: ecpy_hqc_legacy/instruments/drivers/dll/alazar935x.py (hunk 1 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/dll/alazar935x.py:101]: in general I agree, but this is to keep as close as possible to the original python code from AlzarTech

File: ecpy_hqc_legacy/instruments/drivers/driver_tools.py (hunk 2 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/driver_tools.py]: I was having issues with an instrument, and trying more times to connect helped, but in the meantime I found a better solution (communicate through SOCKET rather than LAN). Anyway, changed back.

File: ecpy_hqc_legacy/instruments/drivers/driver_tools.py (hunk 3 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/driver_tools.py:136]: done. Please check the syntax.

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 4 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: ok, please check my syntax

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 5 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: We discussed this. The AWG seems to not clear the output buffer. This must be an instrument bug. So it creates errors when you write read and you use an old answer stored in the buffer to set a parameter

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 6 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py:776]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py (hunk 7 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py]: If I keep this as before, often the PNA doesn't have time to take its trace, and the data looks like the beginning of a trace, and then 0s. I guess that the instrument is not capable of knowing exactly what its sweep time is, since there must be a time overhead to set various parameters before or after the sweep.

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py (hunk 8 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py (hunk 9 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: logged it

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py (hunk 10 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py (hunk 11 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml (hunk 12 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml (hunk 13 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/views/run_awg_views.enaml]: I don't understand, is this ok? (see edit)

File: ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml (hunk 14 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml:73]: added a tooltip. Let me know if it is clear

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py (hunk 15 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py (hunk 16 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py (hunk 17 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: sometimes writing code drives you crazy. Removed.

File: ecpy_hqc_legacy/instruments/drivers/driver_tools.py [hunk 18]
```diff
@@ -133,7 +134,9 @@ def wrapper(self, *args, **kwargs):
...
+                        log = logging.getLogger(__name__)
+                        msg = 'Iterating connection %s/%s' % (i, max_iter)
+                        log.exception(msg)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/driver_tools.py]: We know that this can happen for certain instrument on a regular basis, so I am not sure it is worth to spook the users by logging at the exception level, debug should be sufficient. Furthermore, it may be nice to provide some more infos as currently the user does not even know what instrument issued the log entry. We should at least log the type of the driver even if we cannot get in a reliable way the detailed connection infos.

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py [hunk 19]
```diff
@@ -56,6 +57,27 @@ def clear_sequence(self):
...
 
+    @secure_communication()
+    def set_sequence_pos(self, name, position):
+        """Sets the sequence index position to waveform name.
+
+        """
+        self._AWG.clear_output_buffer()
+        try:
+            current_length = int(self._AWG.ask("SEQuence:LENGth?"))
+        except:
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: NEVER use bare exception as it catches SystemError and KeyboardInterrupt which is not a good idea. To catch general exceptions use ```except Exception:```

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py [hunk 20]
```diff
@@ -425,6 +447,15 @@ def reopen_connection(self):
             except VisaIOError:
                 break
 
+    def clear_output_buffer(self):
+        self.timeout = 100
+        while True:
+            try:
+                self.read()
+            except VisaIOError:
+                self.timeout = 2000
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: Here you should restore the previous value not set an arbitrary one.

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py [hunk 21]
```diff
@@ -453,6 +484,68 @@ def to_send(self, name, waveform):
         self._driver.write_binary_values(header, waveform, datatype='B')
         self.write('*WAI')
 
+    @secure_communication()
+    def clear_sequence(self):
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py:513]: Bunch of missing docstrings. By the way docstrings should start with a capital and finish with a blank line.

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 5 above)
Comment (MatthieuDartiailh) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: This is a known bug of the *CLS command not cleaning the output buffer. However this should only matter after recovering of an error as under normal operation the buffer should not get corrupted. Which is why this method should be refactored inside the clear method (not sure I have the name correctly and should at most be called only when opening the connection or even only when reopening. Otherwise it may mean that the driver is doing something incorrect and not properly reading the buffer at some point.

File: ecpy_hqc_legacy/instruments/drivers/visa_tools.py [hunk 22]
```diff
@@ -120,6 +121,7 @@ def reopen_connection(self):
         as previously.
 
         """
+        self._driver.timeout = 100
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/visa_tools.py]: Why do you alter this setting ? You are overriding the user value.

File: ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py (hunk 7 above)
Comment (MatthieuDartiailh) [reply ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py]: This means that the ```check_operation_complete``` method is not doing its job properly in this case as it should block further execution. What bother me most with this change is that for long sweeps we may end up wasting a lot of time.

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+
+"""
+from __future__ import (division, unicode_literals, print_function,
+                        absolute_import)
+
+from atom.api import (Unicode, Bool, set_default, Enum)
+
+from ecpy.tasks.api import (InstrumentTask, InterfaceableTaskMixin,
+                            validators)
+import logging
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: As per the docs the imports coming from the Python stdlib should come first so please move this import in first position.

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+
+
+class RunAWGTask(InstrumentTask):
+    """ Task to set AWG run mode
+
+    """
+    # switch to choose the AWG run mode: on or off
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: For sphinx to properly pick up those description you need to use ```#:```. And a capital letter would be nice.

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+
+        if switch == 'On' or switch == 1:
+            self.driver.running = 1
+            self.write_in_database('output', 1)
+        else:
+            self.driver.running = 0
+            self.write_in_database('output', 0)
+        log = logging.getLogger(__name__)
+        msg = 'AWG running state OK'
+        log.info(msg)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: debug level is sufficient here.

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py [hunk 26]
```diff
@@ -14,7 +14,7 @@
 
 import numbers
 import numpy as np
-from atom.api import (Bool, Unicode, Enum, set_default)
+from atom.api import (Bool, Unicode, Enum, set_default, Int)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: Int appears to be unused, please remove it.

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py [hunk 27]
```diff
@@ -37,6 +37,9 @@ class DemodSPTask(InstrumentTask):
     average = Enum('No avg', 'Avg before demod',
                    'Avg after demod').tag(pref=True)
 
+    # number of loops in the pulse sequence
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: Same ```#: ```

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py [hunk 28]
```diff
@@ -14,7 +14,7 @@
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py:14]: I trust that none of this changes affect the current behavior.

File: ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml [hunk 29]
```diff
@@ -69,6 +69,17 @@ enamldef DemodSpView(InstrView): view:
...
+    QtLineCompleter: num_loop_val:
+        text := task.num_loop
+        entries_updater << task.list_accessible_database_entries
+        tool_tip = ('Number of loops in your pulse sequence \n '
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml:77]: You can append the standard tool tip for completion after your custom one.

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py [hunk 30]
```diff
@@ -284,6 +284,9 @@ class SaveFileTask(SimpleTask):
     #: Column indices identified as arrays.
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: Could you describe the typical use case for this ? I guess it is for the card.

**Review by MatthieuDartiailh — CHANGES_REQUESTED** [review]

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 5 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: Ok I will put it only in open and reopen, but I think this is what I did initially, and I still had this problem.

Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: I just looked at the code again. The clear_output_buffer is hard coded into reopen_connection, and this doesn't solve the problem. So I am leaving it as it is.

File: ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py (hunk 7 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/pna_tasks.py]: Ok fine I put it back. The fact is this doesn't work, so it needs to be changed locally. maybe it can be a parameter in the task which the user can set.

File: ecpy_hqc_legacy/instruments/drivers/driver_tools.py (hunk 18 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/driver_tools.py]: ok changed to debug, and added driver info

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 19 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: ok

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 20 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py]: ok

File: ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py (hunk 21 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py:513]: ok

File: ecpy_hqc_legacy/instruments/drivers/visa_tools.py (hunk 22 above)
Comment (leghtas) [reply ecpy_hqc_legacy/instruments/drivers/visa_tools.py]: leftovers from debugging the SGS instrument. I put it back

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py (hunk 23 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py (hunk 24 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py (hunk 25 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/run_awg.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py (hunk 26 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py (hunk 27 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py]: ok

File: ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml (hunk 29 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/instr/views/spdev_views.enaml:77]: ok

File: ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py (hunk 30 above)
Comment (leghtas) [reply ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py]: ok

File: ecpy_hqc_legacy/instruments/drivers/driver_tools.py [hunk 31]
```diff
@@ -135,8 +135,9 @@ def wrapper(self, *args, **kwargs):
                         raise
                     else:
                         log = logging.getLogger(__name__)
-                        msg = 'Iterating connection %s/%s' % (i, max_iter)
-                        log.exception(msg)
+                        msg = 'Iterating connection %s/%s ' % (i, max_iter)
```
Comment (MatthieuDartiailh) [inline ecpy_hqc_legacy/instruments/drivers/driver_tools.py]: This could be written more cleanly as:
```python
msg = ('Iterating connection %s/%s '
            'for instrument %s')
log.debug(msg, (i, max_iter, type(self).__name__)
```
This has the added advantage that the message will be formatted only if a logger actually handle it (here we don't care but just so that you know). Furthermore the PyVisa driver will not help the user to know what is going on so here I pass the name of the class so that at least it can know what kind of instrument is behaving badly.

**Comment by MatthieuDartiailh** [comment]
Good work on the whole. I left one small comment and I will redo a more complete review tomorrow.

**Review by MatthieuDartiailh — APPROVED** [review]
This finally looks good. If anything appears unclear to the users we can always iterate later. May I merge @leghtas  ?

**Comment by leghtas** [comment]
Looks good to me, thanks for your very helpful feedback @MatthieuDartiailh
